File: check_braces.js
const fs = require('fs');
const content = fs.readFileSync('frontend-new/src/pages/AdminDashboard.jsx', 'utf-8');

let stack = [];
for (let i = 0; i < content.length; i++) {
  const char = content[i];
  if (char === '{') {
    stack.push({ char, i });
  } else if (char === '}') {
    if (stack.length === 0) {
      console.log(`Extra '}' at index ${i}`);
    } else {
      const last = stack.pop();
      if (last.char !== '{') {
        console.log(`Mismatched '}' at index ${i}, matches '${last.char}' from index ${last.i}`);
      }
    }
  } else if (char === '(') {
    stack.push({ char, i });
  } else if (char === ')') {
    if (stack.length === 0) {
      console.log(`Extra ')' at index ${i}`);
    } else {
      const last = stack.pop();
      if (last.char !== '(') {
        console.log(`Mismatched ')' at index ${i}, matches '${last.char}' from index ${last.i}`);
      }
    }
  }
}

while (stack.length > 0) {
  const last = stack.pop();
  console.log(`Unclosed '${last.char}' from index ${last.i}`);
}

File: check_braces.py

def check_balance(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    stack = []
    for i, char in enumerate(content):
        if char == '{':
            stack.append(('{', i))
        elif char == '}':
            if not stack:
                print(f"Extra '}}' at index {i}")
            else:
                stack.pop()
        elif char == '(':
            stack.append(('(', i))
        elif char == ')':
            if not stack:
                print(f"Extra ')' at index {i}")
            else:
                op, pos = stack.pop()
                if op != '(':
                    print(f"Mismatched ')' at index {i}, matches '{op}' from index {pos}")
    
    while stack:
        op, pos = stack.pop()
        print(f"Unclosed '{op}' from index {pos}")

check_balance('frontend-new/src/pages/AdminDashboard.jsx')
